packages/components/utils/src/useKeyboard/useKeyboard.ts

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useCallback, MutableRefObject } from 'react';
+
+export interface UseKeyboardProps {
+  /**
+   * @description object of key names and handlers defines which key to look for i.e. `ArrowUp`, `Escape`, `Shift`
+   * value is a callback function to be called when key matches
+   */
+  keys: {
+    [key: string]: (e: KeyboardEvent) => void;
+  };
+  /**
+   * @description defines the attached event type
+   * @default 'keydown'
+   */
+  event?: 'keyup' | 'keypress' | 'keydown';
+
+  /**
+   * @description React reference to attach the event to its current element
+   */
+  ref?: MutableRefObject<HTMLElement>;
+}
+
+/**
+ *
+ * @description hook to attach a handler keyboard event listener to `document` or `HTMLElements` with garbage collection
+ * @example
+ * useKeyboard({
+ *  event: 'keydown', // Optional, default is `keydown`
+ *  ref: yourReactRef, // Optional, by default event is attached to document
+ *  keys: {
+ *    ArrowUp: (e) => handleArrowUp(e),
+ *    Tab: (e) => handleTab(e)
+ *  }
+ * })
+ */
+export const useKeyboard = (props: UseKeyboardProps) => {
+  const { ref, keys, event = 'keydown' } = props;
+  let element: HTMLElement | Document = document;
+
+  const handleKeyEvent = useCallback(
+    (e) => {
+      const isKey = Object.prototype.hasOwnProperty.call(keys, e.key);
+      if (isKey) {
+        keys[e.key](e);
+      }
+    },
+    [keys],
+  );
+
+  useEffect(() => {
+    if (ref) {
```

Comment:
Since ref is a mutable object it will always exist, so always `true`. You might rather check `ref.current`